openmdao/core/component.py

```diff
@@ -1259,6 +1259,10 @@ def _declare_partials(self, of, wrt, dct):
                 meta['cols'] = cols
                 meta['shape'] = shape = (abs2meta[abs_key[0]]['size'], abs2meta[abs_key[1]]['size'])
 
+                if abs2meta[abs_key[0]]['size'] == 0:
+                    msg = '{}: Is an array of size 0'
+                    raise ValueError(msg.format(self.msginfo))
+
```

Comment:
To clarify, are we allowed to have zero-length arrays as component inputs and outputs as long as we don't delcare partial derivatives for them explicitly?

Reply:
That's correct